docs/help/contributor/style-guide/documenting-code.md

```diff
@@ -0,0 +1,212 @@
+# Documenting Code
+
+??? tip "Looking for a template?"
+    //todo
+
+## Documenting Commands
+
+>**The standard line is “X the Y by running:”**
+
+>It meets these criteria:
+
+>* Explicitly mentions running the command (this isn’t always obvious)
+* **Uses “run”** (and not “type”, “execute”, etc -- we want consistency)
+* Starts with the key information that describes the command, e.g. “To do X...enter Y command:”, “Do X by entering Y command:”
+* As short as possible
+
+> If you must deviate from the standard line, ensure you still meet the above criteria.
+
+=== ":white_check_mark: Correct"
+    Create the service by running:
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+=== ":no_entry: Incorrect"
+    **Example 1:**
+
+    Create the service:
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+    **Example 2:**
+
+    Run:
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+    **Example 3:**
+
+    Run the command below to create a service:
+
+    ```bash
+    kn create service <service-name>
+    ```
+    Where `<service-name>` is the name of your Knative Service.
+
+
+## Documenting YAML
+>Use **kubectl apply** for files/objects that the user creates — it works for both “create” and “update”, and the source of truth is their local files.
+Use **kubectl edit** for files which are shipped as part of the Knative software, like the serving/eventing ConfigMaps.
+
+=== ":white_check_mark: Correct"
+
+    Creating a new file:
+    ```
+    kubectl apply -f - <<EOF
+    # code
+    EOF
+    ```
+
+    Editing a file:
+    ```
+    kubectl -n <namespace> edit cm <filename>
+    ```
+
+=== ":no_entry: Incorrect"
+    ```
+    cat <<EOF | kubectl create -f -
+    # code
+    EOF
+    ```
+
+
+## Words requiring code formatting
+Use code formatting to indicate special purpose text. Apply code formatting to the following content:
+
+* Text that the user enters into a field
+* Filenames and path names
+* Configuration file content
+* Sections of code
+* Programming examples
+* Command line text
+* Environment variable text
```

Comment:
Feels like it needs to be more specific / have examples, e.g. "Configuration file content" and "Programming examples" to me could mean anything?

Reply:
I think maybe I'll replace this with something like: "anything that could be constituted as "Anything a user would insert into code they are going to run" or "CLI Output" 

I feel like that covers everything?

Reply:
I think it's enough for now to just say filenames and CLI commands and add any additional specific stuff later, afaik that's all it's for right now?

Reply:
Maybe the k8s guide covers this well enough for us to leave it out entirely actually? https://kubernetes.io/docs/contribute/style/style-guide/#use-code-style-for-filenames-directories-and-paths

Reply:
I am trying to make our Style Guide into our own standard to prevent piecemeal bits from different styles. Going to document filenames, CLI commands like you recommended